pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -552,8 +552,8 @@ cpdef ensure_object(object arr):
             return arr
         else:
             return arr.astype(np.object_)
-    elif hasattr(arr, 'asobject'):
-        return arr.asobject
```

Comment:
see what effect of removing this does

Reply:
Two tests fail, both at ``pandas\tests\test_categorical.py:811`` (parameterized tests, of categorical of dtype='timedelta64[h]' and is_ordered=True).